homeassistant/components/dublin_bus_transport/sensor.py

```diff
@@ -45,37 +46,29 @@
 )
 
 
-def due_in_minutes(timestamp):
-    """Get the time in minutes from a timestamp.
-
-    The timestamp should be in the format day/month/year hour/minute/second
-    """
-    diff = datetime.strptime(timestamp, "%d/%m/%Y %H:%M:%S") - dt_util.now().replace(
-        tzinfo=None
-    )
-
-    return str(int(diff.total_seconds() / 60))
-
-
 def setup_platform(hass, config, add_entities, discovery_info=None):
     """Set up the Dublin public transport sensor."""
     name = config[CONF_NAME]
     stop = config[CONF_STOP_ID]
     route = config[CONF_ROUTE]
 
-    data = PublicTransportData(stop, route)
-    add_entities([DublinPublicTransportSensor(data, stop, route, name)], True)
+    try:
+        mybus = DublinBusRTPI(stop, route)
+    except APIError:
+        raise PlatformNotReady
```

Comment:
Since you are only creating the object here, no API is called so the lib will never raise an `APIError` here, so drop the try/except in the setup.

Reply:
you are right about the api, why not using except as Exception instead ?

```
    try:
        mybus = DublinBusRTPI(stop, route)
    except Exception:
        raise PlatformNotReady
```

Reply:
Since there is nothing happening here other than initializing the object, there is no need for a try/except

Reply:
true, but even this can fail under some circumstances

Reply:
The only way this can fail is if `stop`/`route`/`DublinBusRTPI` is missing, and we would not get to this point if they were.

Reply:
okay, makes sense :)  I will drop the `try` block